### ISSUE TITLE
Outdated reference to Microsoft.Web.Infrastructure

### ISSUE THREAD

**Issue by dcorriveau-omniscient** [simpleinjector/SimpleInjector#170]
It seems `Microsoft.Web.Infrastructure` is now a Nuget package, so the nuspec file for `SimpleInjector.Integration.Web` should probably be updated from:
```xml
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.Web.Infrastructure" targetFramework=".NETFramework4.0" />
    </frameworkAssemblies>
```
to:
```xml
    <dependencies>
      <dependency id="Microsoft.Web.Infrastructure" version="1.0.0" />
    </dependencies>
```
see: fsprojects/Paket#1393

Thanks.

**Comment by dotnetjunkie** [comment]
Are you experiencing any problems with the GAC reference?

**Comment by dcorriveau-omniscient** [comment]
As far as I can tell it's only installed in GAC when some tools are also installed (probably some components of Visual Studio) which means it doesn't work on some environments, for example on a Server running my CI.

**Comment by dotnetjunkie** [comment]
The integration package exists for years and many developers use is in production, which means that full .NET 4.0 installations should contain this DLL. Is your build server configured with .NET 4.0-client perhaps?

**Comment by dcorriveau-omniscient** [comment]
Actually it doesn't seem to work on a Win10 dev machine either...
If I run `gacutil -l` it's not listed, I'm guessing it should be ?
It's not in the C:\Windows\Microsoft.NET\assembly folders.
The assembly is found in various other locations on that W10 machine though, such as "C:\Program Files (x86)\Microsoft ASP.NET\ASP.NET MVC 4\..." or "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\Web\Razor\v2.0"
Not sure what I'm doing wrong ?

**Comment by dcorriveau-omniscient** [comment]
Just tested gacutil -l on another dev box and he did have it in the GAC, so I'll have to find out why I don't (find what exactly should register it there).

**Comment by dcorriveau-omniscient** [comment]
After looking at a few developers's box, it seems consistent that machines on WIN10 using VS Community don't have Microsoft.Web.Infrastructure registered in the GAC. So it does make sense that it worked in the past, but I'm suspecting more developers will report this. In any case I can still make it work by installing the required tools (even though it's not ideal), I'll let you determine what's best for SimpleInjector. Thanks !

**Comment by TheBigRic** [comment]
After some investigation it seems that Microsoft.Web.Infrastructure is a NuGet Package since .Net 4.5. I think it causes almost no troubles because a lot of other packages depend on this also and therefore it will be probably added as a NuGet package anyway.

Means that Simple Injector should reference the NuGet package IMO.

**Comment by dotnetjunkie** [comment]
> it seems that Microsoft.Web.Infrastructure is a NuGet Package since .Net 4.5. I 

Is that nuget package compatible with .NET 4.0, because the integration package is .NET 4.0.

**Comment by Selvaticus** [comment]
Hi,

Right now I'm having trouble adding the `SimpleInjector.MVC3` because of this, as I don't have the Microsoft.Web.Infrastructure on the GAC but on the NugetPackages (and I would like to keep it that way, specially because of CI/CD server).

As far as I know, all version of the .Net framework 4.0 and above, are backward compatible as they use the same CLR version, this would mean that applications built for 4.0 can run on 4.5, the reverse will not be true if the code uses new features only introduced on latter versions. [MSDN](https://msdn.microsoft.com/en-us/library/bb822049(v=vs.110).aspx#Anchor_0). This will all come down to which features that Nuget packages uses or not.

IMO, as the .Net teams move their code more and more to a OS agnostic architucture (with ASP.NET 5) dependencies on Nuget is the way to go, as one could deploy to OS without GAC or similar functionality.

Any solution to this problem will be greatly appreciated.

**Comment by dotnetjunkie** [comment]
The [SimpleInjector.MVC3 NuGet package](https://www.nuget.org/packages/SimpleInjector.MVC3/) is the quick start package. All this package does is injecting a [specific code file](https://github.com/simpleinjector/SimpleInjector/blob/master/src/BuildTools/Nuget/.NET/SimpleInjector.Integration.Web.Mvc.QuickStart/content/App_Start/SimpleInjectorInitializer.cs.pp) into your project and link to the appropriate NuGet packages that contain the actual assemblies.

When the `SimpleInjector.MVC3` package gives you trouble, you can simply ignore it and reference the [SimpleInjector.Integration.Web.Mvc](https://www.nuget.org/packages/SimpleInjector.Integration.Web.Mvc/) and use the integration guide as described [here](https://simpleinjector.readthedocs.org/en/latest/mvcintegration.html).

Rest assured that we will fix this problem in the future.

**Comment by dotnetjunkie** [comment]
Hi @Selvaticus, 

the [SimpleInjector.MVC3 NuGet package](https://www.nuget.org/packages/SimpleInjector.MVC3/) is the quick start package. All this package does is injecting a [specific code file](https://github.com/simpleinjector/SimpleInjector/blob/master/src/BuildTools/Nuget/.NET/SimpleInjector.Integration.Web.Mvc.QuickStart/content/App_Start/SimpleInjectorInitializer.cs.pp) into your project and link to the appropriate NuGet packages that contain the actual assemblies.

When the `SimpleInjector.MVC3` package gives you trouble, you can simply ignore it and reference the [SimpleInjector.Integration.Web.Mvc](https://www.nuget.org/packages/SimpleInjector.Integration.Web.Mvc/) and use the integration guide as described [here](https://simpleinjector.readthedocs.org/en/latest/mvcintegration.html).

Rest assured that we will fix this problem in the future.

**Comment by dotnetjunkie** [comment]
Fixed in v3.1.3.

**Comment by leonardolb** [comment]
Hi! im having issues running precompile view.

it seems that even with newer Microsoft.Web.Infrastructure installed, it requires version 1.0.0.

could you update this reference?
